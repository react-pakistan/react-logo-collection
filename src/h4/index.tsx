import styled, { ThemedStyledProps } from 'styled-components';
import { ITheme } from '../theme';

export const H4 = styled.h4<IH4Props>`
  margin: 0;
  padding: 0;
  font-style: normal;
  font-stretch: normal;
  font-family: ${(props : ThemedStyledProps<IH4Props, ITheme>) : string =>
    props.fontFamily || props.theme.fontFamilyH4};
  font-size: ${(props : ThemedStyledProps<IH4Props, ITheme>) : string =>
    props.fontSize || props.theme.fontSizeH4};
  font-weight: ${(props : ThemedStyledProps<IH4Props, ITheme>) : number =>
    props.fontWeight || props.theme.fontWeightH4};
  line-height: ${(props : ThemedStyledProps<IH4Props, ITheme>) : string =>
    props.lineHeight || props.theme.lineHeightH4};
  letter-spacing: ${(props : ThemedStyledProps<IH4Props, ITheme>) : string =>
    props.letterSpacing || props.theme.letterSpacingH4};
  text-align: ${(props : ThemedStyledProps<IH4Props, ITheme>) : string =>
    props.textAlign || props.theme.textAlignH4};
  text-transform: ${(props : ThemedStyledProps<IH4Props, ITheme>) : string =>
    props.textTransform || props.theme.textTransformH4};
  color: ${(props : ThemedStyledProps<IH4Props, ITheme>) : string =>
    props.color || props.theme.colorH4}
`;

export interface IH4Props {
  /**
   * Font Family property for H1 tag
   * @default Montserrat
   */
  fontFamily : string;
  /**
   * Font Size property for H1 tag
   * @default ThemeH4
   */
  fontSize? : string;
  /**
   * Font Weight property for H1 tag
   * @default ThemeH4
   */
  fontWeight? : number;
  /**
   * Line Height proprty for H1 tag
   * @default ThemeH4
   */
  lineHeight? : string;
  /**
   * Letter Spacing property for H1 tag
   * @default ThemeH4
   */
  letterSpacing? : string;
  /**
   * Text Align property for H1 tag
   * @default ThemeH4
   */
  textAlign? : string;
  /**
   * Text Transform property for H1 tag
   * @default ThemeH4
   */
  textTransform? : string;
  /**
   * Color property for H1 tag
   * @default ThemeH4
   */
  color? : string;
}
