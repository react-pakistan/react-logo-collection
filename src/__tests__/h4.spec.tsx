import React from 'react';
import { shallow } from 'enzyme';
import { H4 } from '../h4';
import { h4Props } from '../__stories__/components/h4.story';

describe('Testing H4 component', () => {
  it('matches H4 snapshot', () => {
    const wrapper = shallow(<H4 {...h4Props()}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
