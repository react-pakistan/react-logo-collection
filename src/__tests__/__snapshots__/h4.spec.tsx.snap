// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing H4 component matches H4 snapshot 1`] = `ShallowWrapper {}`;
